### PR TITLE
Simplify the logic for static flight response generation

### PR DESCRIPTION
**Pull request by shuding**
We used to generate the flight page data using a stream which is totally overhead because in that scenario we don't need to stream. This PR simplifies it to just use a string variable.

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: packages/next/server/render.tsx [hunk 1]
```diff
@@ -321,88 +321,80 @@ function checkRedirectValues(
...
+    const forwardReader = forwardStream.getReader()
+    const inlinedDataWriter = inlinedDataWritable.getWriter()
+
+    function process() {
+      forwardReader.read().then(({ done, value }) => {
+        if (!bootstrapped) {
+          bootstrapped = true
+          inlinedDataWriter.write(
+            encodeText(
+              `<script>(self.__next_s=self.__next_s||[]).push(${JSON.stringify([
```
Comment (huozhi) [inline packages/next/server/render.tsx:352]: here we need the escpaed string, need to resolve the conflicts with canary

File: packages/next/server/view-render.tsx [hunk 2]
```diff
@@ -98,57 +98,55 @@ function preloadDataFetchingRecord(
...
+    let bootstrapped = false
+    const forwardReader = forwardStream.getReader()
+    const writer = writable.getWriter()
+    function process() {
+      forwardReader.read().then(({ done, value }) => {
+        if (!bootstrapped) {
+          bootstrapped = true
+          writer.write(
+            encodeText(
+              `<script>(self.__next_s=self.__next_s||[]).push(${JSON.stringify([
```
Comment (huozhi) [inline packages/next/server/view-render.tsx:121]: shall we update JSON.stringify with escapeJSONForFlightScript as well?

Comment (shuding) [reply packages/next/server/view-render.tsx:121]: Yeah probably in another PR. I only applied #36989 here.

Comment (shuding) [reply packages/next/server/view-render.tsx:121]: Actually this branch is out-of-date and checks will be started again anyways. I can push the change here.

**Review by huozhi — APPROVED** [review]

**Pull request merged**